### PR TITLE
Transactional Messaging System Javadocs

### PR DESCRIPTION
**Pull request by awholegunch**
Edits to Javadocs, added @Beta tags to Exception classes, edited exception messages.

Running a complete doc build: http://builds.cask.co/browse/CDAP-DBT6-1

Page of interest: _to be completed_

### CODE REVIEW

File: cdap-api/src/main/java/co/cask/cdap/api/messaging/MessagePublisher.java [hunk 1]
```diff
@@ -27,35 +27,37 @@
...
 /**
- * Provides access to publish messages to the transactional messaging system.
- * Note that for instances acquired through the {@link MessagingContext#getMessagePublisher()} method and
- * when any of the {@code publish} methods is called within a transactional context,
- * exceptions may not be thrown right away to indicate the error, but rather at the transaction commit time.
- * If {@link Transactional#execute(TxRunnable)} was used to execute the transaction, exception will always be wrapped
- * inside a {@link TransactionFailureException}.
+ * Provides message publishing functions of the Transactional Messaging System.
+  * <p>
+ * Note that for instances acquired through either the {@link MessagingContext#getMessagePublisher()}
```
Comment (chtyim) [inline cdap-api/src/main/java/co/cask/cdap/api/messaging/MessagePublisher.java]: There is no "either or". What it meant was, if you get an instance of `MessagePublisher` through the `MessagingContext.getMessagePublish()` method, any call to the `publish` method in that instance will publish transactionally if called from a transactional context. If called outside of a transactional context, then the publish is non-transactional.

Comment (awholegunch) [reply cdap-api/src/main/java/co/cask/cdap/api/messaging/MessagePublisher.java]: Revised.

**Comment by chtyim** [comment]
LGTM

**Comment by chtyim** [comment]
LGTM

**Pull request merged**